src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,211 @@
+- `HazelcastJsonValue` objects cannot be queried at the moment. This limitation will be
+removed in future releases
+
+Field names are case-sensitive.
+
+If the `IMap` does not have local entries on the member where the query is started,
+then the list of fields cannot be resolved, and an exception is thrown.
+
+We refer to the key and the value as *top-level* objects. The key is exposed as
+`__key` field, and the value is exposed as `this` field.
+
+If a top-level object does not have nested fields, then it is exposed as a normal
+field. Otherwise, it is exposed a *hidden field*. Hidden fields behave similarly
+to normal fields, but are not returned by `SELECT * ...` queries.
+
+If both the key and the value have fields with the same name, then the field of the
+value will be used.
+
+==== Indexes
+
+The engine could use `IMap` indexes to speed up the execution of certain queries.
+
+`SORTED` and `HASH` indexes are supported.
+
+==== HD
+
+`IMap` objects with the `InMemoryFormat.NATIVE` could only be queried if they have
+at least one index, and the property `hazelcast.hd.global.index.enabled` is set to
+`true`.
+
+=== Supported Features
+
+==== Data Types
+
+The following SQL types are currently supported.
+
+General:
+
+- `BOOLEAN` (`java.lang.Boolean`)
+- `VARCHAR` (`java.lang.String`, `java.lang.Character`)
+- `OBJECT`
+
+Numeric:
+
+- `TINYINT` (`java.lang.Byte`)
+- `SMALLINT` (`java.lang.Short`)
+- `INTEGER` (`java.lang.Integer`)
+- `BIGINT` (`java.lang.Long`)
+- `DECIMAL` (`java.math.BigDecimal`, `java.math.BigInteger`)
+- `REAL` (`java.lang.Float`)
+- `DOUBLE` (`java.lang.Double`)
+
+Temporal:
+
+- `DATE` (`java.time.LocalDate`)
+- `TIME` (`java.time.LocalTime`)
+- `TIMESTAMP` (`java.time.LocalDateTime`)
+- `TIMESTAMP_WITH_TIME_ZONE` (`java.time.OffsetDateTime`, `java.time.ZonedDateTime`,
+`java.time.Instant`, `java.util.Date`, `java.util.Calendar`)
+
+==== Operators
+
+It is possible to execute project-filter queries in the form
+`SELECT ... FROM <map_name> WHERE ...` on a single map.
+
+The support for the following operators are planned for future releases:
+
+- `ORDER BY`
+- `GROUP BY`
+- Joins
+- Set operators (`UNION`, `INTERSECT`, `MINUS`)
+- Subqueries
+
+==== Functions
+
+The following functions are currently supported.
+
+- Binary functions: `+`, `-`, `*`, `/`
+- Comparison predicates: `>`, `>=`, `<`, `<=`, `=`, `!=`, `<>`
+- Logical predicates: `AND`, `OR`, `NOT`
+- `IS` predicates: `IS [NOT] NULL`, `IS [NOT] TRUE`, `IS [NOT] FALSE`
+- `CAST` function
+- Math functions: `ABS`, `ACOS`, `ASIN`, `ATAN`, `CEIL`, `COS`, `COT`,
+`DEGREES`, `EXP`, `FLOOR`, `LN`, `LOG10`, `RADIANS`, `RAND`, `ROUND`,
+`SIGN`, `SIN`, `TAN`, `TRUNCATE`
+- String functions: `||` (concatenation), `ASCII`, `INITCAP`, `LENGTH`,
+`LIKE`, `LOWER`, `LTRIM`, `RTRIM`, `SUBSTRING`, `TRIM`, `UPPER`
+
+More functions will be added in future releases.
+
+=== Clients
+
+SQL queries could be executed from Java clients using the same API as with members.
```

Comment:
Short example please. Also the headline might be "Execute SQL from clients" or something like that

Reply:
The API is exactly the same, therefore it will be just duplication. We do not do this normally for other sections. Also, we do not mention clients unless there is some specific treatment. Therefore, it is better to remove this section altogether - it doesn't add any value to the documentation currently.